### PR TITLE
Use types, decode, encode from wallet-common

### PR DESCRIPTION
**Pull request by meriadec**
:warning: [this pull request on wallet-common](https://github.com/LedgerHQ/ledger-wallet-common/pull/3) needs to be merged first. (edit: it's ok now)

**BONUS** you will have to remove your accounts.json file

### CODE REVIEW

File: src/components/OperationsList/stories.js [hunk 1]
```diff
@@ -39,7 +37,7 @@ const operations = [
     address: '5c6ea1716520c7d6e038d36a3223faced3c',
     hash: '5c6ea1716520c7d6e038d36a3223faced3c4b8f7ffb69d9fb5bd527d562fdb62',
     amount: 130000000,
-    receivedAt: '2018-01-09T16:03:52Z',
+    date: '2018-01-09T16:03:52Z',
```
Comment (gre) [inline src/components/OperationsList/stories.js]: should be `new Date(..`

File: src/helpers/btc.js [hunk 2]
```diff
@@ -31,21 +30,25 @@ export function computeOperation(addresses: Array<string>) {
...
+      accountId: 'abcd',
+      blockHeight: 0,
     }
   }
 }
 
 export function getBalanceByDay(operations: Operation[]) {
   const txsByDate = groupBy(operations, tx => {
-    const [date] = new Date(tx.receivedAt).toISOString().split('T')
+    const [date] = new Date(tx.date).toISOString().split('T')
```
Comment (gre) [inline src/helpers/btc.js]: tx.date is already a Date ?

File: src/reducers/accounts.js [hunk 3]
```diff
@@ -1,42 +1,34 @@
...
 
 export type AccountsState = Account[]
 const state: AccountsState = []
 
+const accountModel = createAccountModel()
+
 function orderAccountsOperations(account: Account) {
   const { operations } = account
-  operations.sort((a, b) => new Date(b.receivedAt) - new Date(a.receivedAt))
+  operations.sort((a, b) => new Date(b.date) - new Date(a.date))
```
Comment (gre) [inline src/reducers/accounts.js]: same?

btw I think `orderAccountsOperations` can be dropped, we assume account.operations are always correctly sorted in all our current algorithm. maybe you just need to insert at the correct place when you add a new op?

File: src/reducers/accounts.js [hunk 4]
```diff
@@ -98,31 +90,19 @@ export function canCreateAccount(state: State): boolean {
...
-    currency: getCurrencyByCoinType(account.coinType),
-    name: account.name || `${key}`,
-    unit: account.unit || getDefaultUnitByCoinType(account.coinType),
-  }))
+// Yeah. `any` should be `AccountRaw[]` but it can also be a map
+// of wrapped accounts. And as flow is apparently incapable of doing
+// such a simple thing, let's put any, right? I don't care.
+export function serializeAccounts(accounts: any): Account[] {
+  // ensure that accounts are always wrapped in data key
+  if (accounts.length && !accounts[0].data) {
```
Comment (gre) [inline src/reducers/accounts.js]: probably we could `try {} catch (){}` on the decode code & if it fails, fallback on returning `[]` , it's a safe thing to do if data happen to be corrupted in some way & so it does not brick the loader (i'm sure there are many ways things can go south & maybe we should assume the file we import can't be 100% trust. obviously we'll try to cover all cases via the migrations functions)

File: src/components/OperationsList/stories.js (hunk 1 above)
Comment (meriadec) [reply src/components/OperationsList/stories.js]: true

File: src/reducers/accounts.js (hunk 3 above)
Comment (meriadec) [reply src/reducers/accounts.js]: good point, but let's move that input to a future commit, as it's not directly related to types changes

File: src/helpers/btc.js (hunk 2 above)
Comment (gre) [reply src/helpers/btc.js]: OH ok

File: src/reducers/accounts.js (hunk 4 above)
Comment (gre) [reply src/reducers/accounts.js]: probably we will keep it like this and a bit later you can simplify this code to no longer support migrating old dev version ;)

**Pull request merged**